triggerbot/tree_watcher.py

```diff
@@ -306,35 +310,29 @@ def attempt_triggers(self, branch, rev, builder, count, seen=0, attempt=0):
 
             self.log.warning('Will re-attempt')
             tm = Timer(90, self.attempt_triggers,
-                       args=[branch, rev, builder, count, seen, attempt + 1])
+                       args=[repo_name, rev, builder, count, seen, attempt + 1])
             tm.start()
             # Assume some subsequent attempt will be succesful for accounting
             # purposes.
             return count
 
-        self._rebuild(build_url, payload)
         return count
 
-    def _get_ids_for_rev(self, branch, rev, builder):
+    def _get_ids_for_rev(self, repo_name, rev, builder):
         # Get the request or build id associated with the given branch/rev/builder,
         # if any.
-        root_url = 'https://secure.pub.build.mozilla.org/buildapi/self-serve'
-
-        # First find the build_id for the job to rebuild
-        build_info_url = '%s/%s/rev/%s?format=json' % (root_url, branch, rev)
-        info_req = requests.get(build_info_url,
-                                headers={'Accept': 'application/json'},
-                                auth=self.auth)
         found_buildid = None
         found_requestid = None
         builder_total, rev_total = 0, 0
-
+        # First find the build_id for the job to rebuild
         try:
-            results = info_req.json()
+            results = QUERY_SOURCE.get_all_jobs(repo_name, rev)
         except ValueError:
             self.log.error('Received an unexpected ValueError when retrieving '
                            'information about %s from buildapi.' % rev)
-            self.log.error('Request status: %d' % info_req.status_code)
+            if results == []:
+                self.log.error('Request status not in 200.')
```

Comment:
I see how this is a best estimate.
Another approach would be using "ValueError, e" and output the info for it.
In any case, I believe it is not important enough to get the exact same behaviour.